src/edu/csus/ecs/pc2/ui/MultipleDataSetPane.java

```diff
@@ -478,7 +479,20 @@ protected void loadDataFiles() {
             dump(problemDataFiles, "debug before load");
         }
         try {
+
+            String sampleBaseDirectoryName = new File(baseDirectoryName).getParent() + File.separator +"sample";
+            boolean ishere = new File(sampleBaseDirectoryName).isDirectory();
+            
+            if (loadSamplesFirstCheckbox.isSelected() && new File(sampleBaseDirectoryName).isDirectory()) {
+
+                getLog().info("Loading sample files from sample dir " + sampleBaseDirectoryName);
+
+                // load sample files first 
+                problemDataFiles = loadDataFiles(problem, problemDataFiles, sampleBaseDirectoryName, ".in", ".ans", externalFiles);
+            }
+
             problemDataFiles = loadDataFiles(problem, problemDataFiles, baseDirectoryName, ".in", ".ans", externalFiles);
+   
```

Comment:
Taken together, the above lines of code (486 through 494) have the effect that, if the "Load Sample Data" checkbox for the problem is checked, then the order of the data files will be (1) all data files in the "sample" directory, followed by (2) all data files in the "secret" directory.  I believe it is the case that when a submission is executed, it is THIS order which determines the order in which the data files will be executed (although someone could convince me that this is not correct; i.e., that the `Executable` class does NOT use the "load order" in determining the order in which test data files are to be executed).

However, if I'm correct that `Executable` IS using the order in which the test data files are loaded to determine the order of execution, then the above code does not meet CLICS specifications.

In particular, the CLICS specs require that data files be processed (executed) in lexicographic order of the base filename.   So if we have a `sample` directory containing two files named `firstsample.{in,ans}` and `secondsample.{in,ans}`, and a `secret` directory containing two files named `bruteforce.{in,ans}` and `quick.{in,ans}`, then if `Executable` is indeed using the load order to determine execution order, these files will be executed in the order
```
firstsample
secondsample
bruteforce
quick
```
whereas by the CLICS lexicographic rule they are supposed to be executed in the order
```
bruteforce
firstsample
quick
secondsample
```

Note: do not get hung up on the fact that conceptually it's likely that the judges want "all the samples executed before the secrets".  If that's truly what they want, then according to the CLICS spec, _they have the responsibility of ensuring that all the "sample" files have base filenames which come lexicographically before all the secret files_. (There's even a note in the [ICPC Problem Package Format specification](https://icpc.io/problem-package-format/spec/problem_package_format#test-data) to this effect.) The point is, the above code will fail to load the collection of sample+secret files in lexicographic order.

Reply:
Applying the CLICS filename sorting rule to the filenames in a directory which I think is the intent, rather
than load all samples and judges files then sorting them.   It is my understanding that all other CCS 
use this same method to load/order data files.
As implemented, reads sample filenames, sorts then loads.
Then reads judges filenames sorts, then loads.
Exectuable loops through the Problems/ProblemData files as loaded.

Reply:
Note that the pc2tool to copy/rename the samples into the secret directory uses the same load order 
as implmented in this new code.

Reply:
I have been attempting to pursue this issue.  According to Nicky, DOMjudge uses the same approach you have implemented:  load all `sample` files (in order by filename), _then_ load all `secret` files (in order by file name).  He says their justification is this statement in the CLICS spec:
```
For pass-fail problems, the verdict of a submission is the first non-accepted verdict, where test cases are run in lexicographical order of their full file paths (note that sample comes before secret in this order).
```
However, the CLICS spec _also_ contains the following statement (at https://icpc.io/problem-package-format/spec/problem_package_format#test-data, under "Test Groups"):
```
Test files and groups will be used in lexicographical order on file base name. 
```
So, in one place the spec says "use full path name", while in another it says "use file base name".  Therefore what we have is an ambiguous spec (shocking!)   I am accordingly withdrawing all requests to change the order, until we can get the CLICS group to resolve the ambiguity.  When that's done, if the resolution is to the latter (rather than the former) statement, we'll have to update our code (as will DOMjudge...)